src/main/java/org/generationcp/middleware/api/germplasm/GermplasmService.java

```diff
@@ -28,6 +28,10 @@
 
 public interface GermplasmService {
 
+	Germplasm getGermplasmByGID(Integer gid);
```

Comment:
As much as possible we shouldn't be exposing Germplasm object - the raw Hibernate entity from MW. It seems we are interested in the mgid property so maybe we can just return a map of mgid for all GIDS to code
`Map<Integer, Integer> getGermplasmGroupIdMap`
with gid as key and mgid as map value.

Reply:
I think we should stop returning Maps as much as we can because these functions are not reusable at all. Maps should be created where they are really needed. 
For what he needs he can use GermplasmDto and we have some functions in place returning this object